diag_manager/diag_axis.F90

```diff
@@ -138,9 +138,11 @@ INTEGER FUNCTION diag_axis_init(name, DATA, units, cart_name, long_name, directi
     ENDIF
 
     if (use_modern_diag) then
+      !TODO Passing in the axis_length because of a gnu issue where inside fms_diag_axis_init, the size of DATA
+      !was 2 which was causing the axis_data to not be written correctly...
       diag_axis_init = fms_diag_object%fms_diag_axis_init(name, DATA, units, cart_name, long_name=long_name,&
        & direction=direction, set_name=set_name, edges=edges, Domain=Domain, Domain2=Domain2, DomainU=DomainU, &
-       & aux=aux, req=req, tile_count=tile_count, domain_position=domain_position )
+       & aux=aux, req=req, tile_count=tile_count, domain_position=domain_position, axis_length=size(DATA(:)) )
```

Comment:
@rem1776 @thomas-robinson  FYI i saw this with gnu 9.3. Let me know if you have a better fix for this. 

If I were to print the size(DATA(:)) here it gave me the correct answers, but inside fms_diag_axis_init it was equal to 2